lib/dry/validation/values.rb

```diff
@@ -70,6 +70,8 @@ def key?(key, hash = data)
             return result
           elsif e.is_a?(Symbol) && a.is_a?(Array)
             return false
+          elsif a.is_a?(String) || a.nil?
+            return false
```

Comment:
Why `a.is_a?(String)` was needed? It shouldn't be, at this point all keys should be symbolized.

Reply:
It for the case when root value is a string. I have checked and it is not symbolized.
It covered by this test case
```
  let(:data) do
    {
      name: "Jane",
      address: {city: "Paris", geo: {lat: 1, lon: 2}},
      phones: [123, 431],
      billing_address: nil,
      card: ""
    }
  end

    it "returns false when a nested key is not present and root key is string" do
      expect(values.key?([:card, :not_here])).to be(false)
    end

```

Reply:
Maybe I should check for valid types instead?

```
          elsif !(a.is_a?(Array) || a.is_a?(Hash))
            return false
```

Reply:
@alexxty7 aahh that's right, this can happen! I think it should be handled separately though, so:

```ruby
# ...
elsif a.nil?
  return false
elsif a.is_a?(String)
  return false
# ...
end
```

Reply:
@solnic updated